### ISSUE TITLE
use readr for fwf

### ISSUE THREAD

**Issue by markdanese** [leeper/rio#48]
This package looks great.  Thanks for putting it together.  The relatively new [readr](https://github.com/hadley/readr) package from hadley seems to be the fastest way to read in fixed width files.  You might want to consider using that package/function.  Just some background on things I found when trying to find fast ways of reading [fixed width files.](http://stackoverflow.com/questions/24715894/faster-way-to-read-fixed-width-files-in-r)

**Comment by leeper** [comment]
Thanks, @markdanese. We've discussed this a bit here: #3 and sided with `data.table::fread` for most text-based files at this point. **readr** was a bit new at the time but we're still considering options moving forward. It's not 100% obvious to use readr over data.table, as the functionality differs a bit, but I'm open to arguments either way over at #3 (so I'll close this discussion).

**Comment by leeper** [comment]
@markdanese Sorry, I guess I read this too quickly. Yes, this specific feature is worth considering more in depth.

**Comment by leeper** [comment]
To implement this, I'd like to follow the base `read.fwf()` argument conventions, so we'll need a utility function to convert from standard column/width notation into the `col_positions` argument used by `read_fwf()`. This means passing widths to `fwf_widths()` or allowing someone to directly an `fwf_widths()` or `fwf_positions()` object.

Also need to be careful about passing `col_positions` and `col_types` arguments given non-standard naming (compared to base packages).

**Comment by markdanese** [comment]
For what it is worth, here is what I did in my package that reads NHAMCS files.  You probably have this covered, but on the off chance it is helpful, I will put it below.  (Thanks for your work on the rio package.)

Read in a text file with specs into a specific format (below).  I am reading from a SAS dataset specification, but you might want to leave it to users to generate this file.  Note that description (desc) is not really necessary but was available in the data.  It could be set as an attribute on a column using the labelled class that some packages use.  I just divided the world into character and numeric.  But one could specify a single column that holds variable type information.  Or let `read_fwf()` figure it out.  Since my data has some of this information, I tried to use it.
```
specs <- data.frame(colstart, colstop, varname, char, num, width, desc) # file of database specs
```

Then I generate an input specification file:
```
input_specs <- fwf_positions(specs$colstart, specs$colstop, specs$varname)
```

Then I generate a column type  specification vector (with magrittr operator in there and a wonky `ifelse()` construction):
 ```
coltype <- ifelse(specs$char, "c", 
                 ifelse(specs$num, "d", "i")) %>% 
           paste0(., collapse = "") # one character per column for types
```

Then I read the file (y, in this case):
```
d <- read_fwf(y, input_specs, col_types = coltype, progress = FALSE)
```
